File: app/src/main/java/com/fphoenixcorneae/viewpagertransformer/RotateDownTransformer.java
package com.fphoenixcorneae.viewpagertransformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * Created on 2017/6/30.
 */

public class RotateDownTransformer extends BaseTransformer {

    private static final float ROT_MOD = -15F;

    @Override
    protected void onTransform(View page, float position) {
        float pageWidth = page.getWidth();
        float pageHeight = page.getHeight();
        float rotation = ROT_MOD * position * -1.25F;

        ViewHelper.setPivotX(page, pageWidth * 0.5F);
        ViewHelper.setPivotY(page, pageHeight);
        ViewHelper.setRotation(page, rotation);

    }

    @Override
    protected boolean isPagingEnabled() {
        return true;
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/view_pager_transformer/RotateDownTransformer.kt
package com.fphoenixcorneae.view_pager_transformer

import android.view.View
import com.nineoldandroids.view.ViewHelper

/**
 * @desc 左转一定角度的转换效果
 * @date 2017-06-30
 */
class RotateDownTransformer : AbstractBaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        val pageWidth = page.width.toFloat()
        val pageHeight = page.height.toFloat()
        val rotation = ROT_MOD * position * -1.25f
        ViewHelper.setPivotX(page, pageWidth * 0.5f)
        ViewHelper.setPivotY(page, pageHeight)
        ViewHelper.setRotation(page, rotation)
    }

    override val isPagingEnabled: Boolean = true

    companion object {
        private const val ROT_MOD = -15f
    }
}